relational_operators/DropTableOperator.cpp

```diff
@@ -32,9 +34,21 @@ namespace quickstep {
 bool DropTableOperator::getAllWorkOrders(WorkOrdersContainer *container) {
   if (blocking_dependencies_met_ && !work_generated_) {
     work_generated_ = true;
+
+    std::vector<block_id> relation_blocks(relation_.getBlocksSnapshot());
+
+    // DropTableWorkOrder only drops blocks, if any.
     container->addNormalWorkOrder(
-        new DropTableWorkOrder(relation_.getID(), only_drop_blocks_),
+        new DropTableWorkOrder(std::move(relation_blocks)),
```

Comment:
No, it won't result in a deep copy. 

``new DropTableWorkOrder(relation_blocks)`` -> Here ``relation_blocks`` is passed as a const reference, so no copy has happened so far. 

In the initialization of blocks_ in the ``DropTableWorkOrder``, you are doing a move. Therefore deep copy is out of the question any way. 
```
explicit DropTableWorkOrder(std::vector<block_id> &&blocks)
      : blocks_(std::move(blocks)) {}
```